java/ql/src/semmle/code/java/frameworks/Networking.qll

```diff
@@ -41,3 +41,88 @@ class SocketGetInputStreamMethod extends Method {
     hasNoParameters()
   }
 }
+
+/** Any expresion or call which returns a new URI. */
+abstract class UriCreation extends Top {
```

Comment:
Avoid public `abstract` classes - they come with a lot of semantic baggage and should be considered an advanced level ql feature not to be used lightly.
Instead you can do
```
class UriCreation extends Call {
  UriCreation() {
    this.getCallee().getDeclaringType() instanceof TypeUri and
    (this instanceof ClassInstanceExpr or this.getCallee().hasName("create"))
  }

  Expr getHostArg() { none() }
}
```
You can then define the two subclasses and override `getHostArg` as you currently do in order to keep that defined via dispatch.

Also, you're not using `UriCreation`? Is that an oversight?

Reply:
yes that was an oversight.